nexus/tests/integration_tests/saml.rs

```diff
@@ -0,0 +1,1058 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use nexus_test_utils::http_testing::{AuthnMode, NexusRequest, RequestBuilder};
+use omicron_common::api::external::IdentityMetadataCreateParams;
+use omicron_nexus::authn::silos::{
+    IdentityProviderType, SamlIdentityProvider, SamlLoginPost,
+};
+use omicron_nexus::external_api::console_api;
+use omicron_nexus::external_api::params;
+use omicron_nexus::external_api::views::{self, Silo};
+use omicron_nexus::TestInterfaces;
+
+use http::method::Method;
+use http::StatusCode;
+use nexus_test_utils::resource_helpers::{create_silo, object_create};
+
+use nexus_test_utils::ControlPlaneTestContext;
+use nexus_test_utils_macros::nexus_test;
+
+use httptest::{matchers::*, responders::*, Expectation, Server};
+
+// Valid SAML IdP entity descriptor from https://en.wikipedia.org/wiki/SAML_metadata#Identity_provider_metadata
+// note: no signing keys
+pub const SAML_IDP_DESCRIPTOR: &str = r#"
```

Comment:
Any particular reason this is an inline string vs. the other stuff that's in separate data files?